### PR TITLE
Add EncodeAeson stuff.

### PR DESCRIPTION
**Pull request by adamczykm**
This adds support for EncodeJson equivalent for Aesons. It will most likely be useful for CTL-bridge things.
- [x] replace array with catlist as the number index container
- [x] tests todo

### CODE REVIEW

File: src/Aeson.purs [hunk 1]
```diff
@@ -55,35 +65,36 @@ module Aeson
 
 import Prelude
 
+import Contract.Prelude (Tuple(..), length, sequence)
```
Comment (ngua) [inline src/Aeson.purs:68]: `Contract` imports

File: src/Aeson.purs [hunk 2]
```diff
@@ -475,3 +486,142 @@ instance DecodeAeson a => DecodeAesonField (Maybe a) where
...
+newtype AesonEncoder a = AesonEncoder (State Int a)
+
+derive newtype instance Functor AesonEncoder
+derive newtype instance Apply AesonEncoder
+derive newtype instance Applicative AesonEncoder
+derive newtype instance Bind AesonEncoder
+derive newtype instance Monad AesonEncoder
+
+genNextNumberIndex :: AesonEncoder Int
+genNextNumberIndex = AesonEncoder (get <* modify_ (\s -> s + 1))
```
Comment (ngua) [inline src/Aeson.purs:507]: Personal style opinion: I'd prefer a section here

File: src/Aeson.purs [hunk 3]
```diff
@@ -475,3 +486,142 @@ instance DecodeAeson a => DecodeAesonField (Maybe a) where
...
+
+getCurrentNumberIndex :: AesonEncoder Int
+getCurrentNumberIndex = AesonEncoder get
+
+bumpNumberIndexBy :: Int -> AesonEncoder Unit
+bumpNumberIndexBy i = AesonEncoder (modify_ (\s -> s + i))
```
Comment (ngua) [inline src/Aeson.purs:513]: Here as well

**Review by ngua — CHANGES_REQUESTED** [review]
Please remove the `Contract` imports

We should export the new typeclass from `Contract.Aeson`. We might want to add instances for various Plutus types as well

Are you planning on implementing the tests in this PR as well?

**Comment by adamczykm** [comment]
Yes, I'll add some today.

**Review by ngua — APPROVED** [review]
Looks good. Could you just add a `Contract` export for the new class?

**Review by klntsky — APPROVED** [review]

File: src/Aeson.purs [hunk 4]
```diff
@@ -475,3 +485,143 @@ instance DecodeAeson a => DecodeAesonField (Maybe a) where
...
+newtype AesonEncoder a = AesonEncoder (State Int a)
+
+derive newtype instance Functor AesonEncoder
+derive newtype instance Apply AesonEncoder
+derive newtype instance Applicative AesonEncoder
+derive newtype instance Bind AesonEncoder
+derive newtype instance Monad AesonEncoder
+
+genNextNumberIndex :: AesonEncoder Int
+genNextNumberIndex = AesonEncoder (get <* modify_ ((+) 1))
```
[the remark this answers is not in the document]
Comment (adamczykm) [reply src/Aeson.purs:506]: Ok

**Pull request merged**